### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by kalocsaadi**

### CODE REVIEW

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 1]
```diff
@@ -14,6 +14,8 @@ public class TorpedoStore {
 
   private int torpedoCount = 0;
 
+  private Random generator = new Random();
```
Comment (SzAdamHun) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:17]: Nice, makes the code cleaner.

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 2]
```diff
@@ -30,18 +32,17 @@ public TorpedoStore(int numberOfTorpedos){
 
   public boolean fire(int numberOfTorpedos){
     if(numberOfTorpedos < 1 || numberOfTorpedos > this.torpedoCount){
-      new IllegalArgumentException("numberOfTorpedos");
+      throw new IllegalArgumentException("numberOfTorpedos");
```
Comment (SzAdamHun) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:35]: Very nice.

Comment (kalocsaadi) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:35]: omg crazy

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 3]
```diff
@@ -30,18 +32,17 @@ public TorpedoStore(int numberOfTorpedos){
...
     boolean success = false;
 
     // simulate random overheating of the launcher bay which prevents firing
-    Random generator = new Random();
     double r = generator.nextDouble();
 
     if (r >= FAILURE_RATE) {
       // successful firing
-      this.torpedoCount =- numberOfTorpedos;
+      this.torpedoCount -= numberOfTorpedos;
```
Comment (SzAdamHun) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:45]: Bug fixed. Nice.

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 4]
```diff
@@ -30,18 +32,17 @@ public TorpedoStore(int numberOfTorpedos){
...
   public boolean fire(int numberOfTorpedos){
     if(numberOfTorpedos < 1 || numberOfTorpedos > this.torpedoCount){
-      new IllegalArgumentException("numberOfTorpedos");
+      throw new IllegalArgumentException("numberOfTorpedos");
     }
 
     boolean success = false;
 
     // simulate random overheating of the launcher bay which prevents firing
-    Random generator = new Random();
```
Comment (kalocsaadi) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:39]: insane skill

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 5]
```diff
@@ -30,18 +32,17 @@ public TorpedoStore(int numberOfTorpedos){
...
     double r = generator.nextDouble();
 
     if (r >= FAILURE_RATE) {
       // successful firing
-      this.torpedoCount =- numberOfTorpedos;
```
Comment (kalocsaadi) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:44]: so unexpected

**Review by kalocsaadi — COMMENTED** [review]
Amazing job!

**Review by SzAdamHun — COMMENTED** [review]
Very good changes overall. Keep it up!

**Review by SzAdamHun — APPROVED** [review]
Looks good to me. Approved.

**Pull request merged**